tests/conftest.py

```diff
@@ -0,0 +1,43 @@
+# pylint: disable=redefined-outer-name
+from pathlib import Path
+from contextlib import contextmanager
+from tempfile import NamedTemporaryFile
+import pytest
+from filelock import FileLock
+from solc_select import solc_select
+from slither import Slither
+
+
+@pytest.fixture()
+def solc_binary_path():
+    def inner(version):
+        lock = FileLock(f"{version}.lock", timeout=60)
```

Comment:
This file sticks around which is a little annoying

Reply:
Maybe you can do something like the following to have a global temp dir across all tests, and then create filelocks there: https://github.com/pytest-dev/pytest/issues/1402#issuecomment-186299177

Reply:
Awesome, thanks! I think this is ready to go now that I've made this change https://github.com/crytic/slither/pull/1808/commits/6451285f8f62d7fb24d79d52fa47e7039a2f2b0c